### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by tenhobi**
Implemented following features:

- [x] primitive types (num, int, double, string, bool) + enum
- [x] iterable types
    - [x] iterable
    - [x] list
    - [x] set
    - [x] map
- [x] complex/nested types
- [x] renaming
- [x] custom mapping - functions or const values
- [x] when null default - field
- [x] when null default - target
- [x] constructor parameters (positional, named)
- [x] mapping to target (1. to constructor parameters, 2. to public fields and setters that have not been set)
- [x] mapping from source (public fields or getters of either instance or static)
- [x] selecting the constructor
- [x] works with Equatable
- [x] works with JSON serializable

### CODE REVIEW

File: packages/auto_mappr/lib/generator/auto_mapper_generator.dart [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+/// Code generator to generate implemented mapping classes.
+class AutoMapperGenerator extends GeneratorForAnnotation<AutoMappr> {
+  @override
+  dynamic generateForAnnotatedElement(Element element, ConstantReader annotation, BuildStep buildStep) {
+    if (element is! ClassElement) {
+      throw InvalidGenerationSourceError(
+        '${element.displayName} is not a class and cannot be annotated with @AutoMap',
+        element: element,
+        todo: 'Add AutoMap annotation to a class',
```
Comment (petrnymsa) [inline packages/auto_mappr/lib/generator/auto_mapper_generator.dart:21]: AutoMappr

File: packages/auto_mappr/lib/generator/auto_mapper_generator.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        fieldMappings: fieldMappings,
+        whenSourceIsNullExpression: whenSourceIsNull,
+        constructor: constructor,
+      );
+    }).toList();
+
+    final duplicates = mappers.duplicates;
+    if (duplicates.isNotEmpty) {
+      throw InvalidGenerationSourceError(
+        '@AutoMap has configured duplicated mappings:\n\t${duplicates.join('\n\t')}',
```
Comment (petrnymsa) [inline packages/auto_mappr/lib/generator/auto_mapper_generator.dart:64]: AutoMapper has configured...

File: packages/auto_mappr/lib/generator/auto_mapper_generator.dart [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      );
+    }
+
+    final config = AutoMapperConfig(mappers: mappers);
```
Comment (petrnymsa) [inline packages/auto_mappr/lib/generator/auto_mapper_generator.dart:68]: AutoMapprConfig

File: packages/auto_mappr/lib/models/field_mapping.dart [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  Expression apply(SourceAssignment assignment) {
+    if (ignore) {
+      return assignment.getDefaultValue();
+    }
+
+    if (hasCustomMapping()) {
+      return customExpression!;
+    }
+
+    // if (isRenamed()) {
```
Comment (petrnymsa) [inline packages/auto_mappr/lib/models/field_mapping.dart:61]: cleanup commented code

**Review by petrnymsa — CHANGES_REQUESTED** [review]

File: packages/auto_mappr/lib/generator/auto_mapper_generator.dart (hunk 2 above)
Comment (tenhobi) [reply packages/auto_mappr/lib/generator/auto_mapper_generator.dart:64]: MapType, but yes 👍

File: packages/auto_mappr/lib/models/field_mapping.dart [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    //         refer(_whenNullDefault.referCallString).call([]),
+    //         convertCallExpr,
+    //       );
+    // }
+
+    throw InvalidGenerationSourceError(
+      'FieldMapping for field "$field" from $assignment has ignore=false and target=null',
+      todo: 'Set ignore=true or define custom mapping',
+    );
+  }
```
Comment (tenhobi) [inline packages/auto_mappr/lib/models/field_mapping.dart:85]: Not sure what should be the correct message here now.

Comment (petrnymsa) [reply packages/auto_mappr/lib/models/field_mapping.dart:65]: I think message is ok, it just need some rewording. 

FieldMapping for field "$field" from $assignment has ignore=false no custom mapping.

todo: 'Either set ignore to true or define custom mapping',

**Review by petrnymsa — CHANGES_REQUESTED** [review]

**Review by petrnymsa — APPROVED** [review]

**Review by petrnymsa — APPROVED** [review]

**Review by petrnymsa — APPROVED** [review]

**Pull request merged**